aspnetcore/security/authorization/iard.md

```diff
@@ -0,0 +1,171 @@
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/security/authorization/OldStyleAuthRequirements/Program.cs" highlight="9":::
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/security/authorization/OldStyleAuthRequirements/Controllers/GreetingsController.cs" :::
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/security/authorization/OldStyleAuthRequirements/Authorization/MinimumAgeAuthorizationHandler.cs" highlight="7,19":::
+
+The complete sample is [here](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/security/authorization/OldStyleAuthRequirements) in the [AspNetCore.Docs.Samples](https://github.com/dotnet/AspNetCore.Docs.Samples) repository.
+
+ASP.NET Core 8 introduces the <xref:Microsoft.AspNetCore.Authorization.IAuthorizationRequirementData> interface. The `IAuthorizationRequirementData` interface allows the attribute definition to specify the requirements associated with the authorization policy. Using `IAuthorizationRequirementData`, the preceding custom authorization policy code can be written with fewer lines of code. The updated `Program.cs` file:
+
+```diff
+  using AuthRequirementsData.Authorization;
+  using Microsoft.AspNetCore.Authorization;
+  
+  var builder = WebApplication.CreateBuilder();
+  
+  builder.Services.AddAuthentication().AddJwtBearer();
+  builder.Services.AddAuthorization();
+  builder.Services.AddControllers();
+- builder.Services.AddSingleton<IAuthorizationPolicyProvider, MinimumAgePolicyProvider>();
+  builder.Services.AddSingleton<IAuthorizationHandler, MinimumAgeAuthorizationHandler>();
+  
+  var app = builder.Build();
+  
+  app.MapControllers();
+  
+  app.Run();
+```
+
+The updated `MinimumAgeAuthorizationHandler`:
+
+```diff
+using Microsoft.AspNetCore.Authorization;
+using System.Globalization;
+using System.Security.Claims;
+
+namespace AuthRequirementsData.Authorization;
+
+- class MinimumAgeAuthorizationHandler : AuthorizationHandler<MinimumAgeRequirement>
++ class MinimumAgeAuthorizationHandler : AuthorizationHandler<MinimumAgeAuthorizeAttribute>
+{
+    private readonly ILogger<MinimumAgeAuthorizationHandler> _logger;
+
+    public MinimumAgeAuthorizationHandler(ILogger<MinimumAgeAuthorizationHandler> logger)
+    {
+        _logger = logger;
+    }
+
+    // Check whether a given MinimumAgeRequirement is satisfied or not for a particular
+    // context
+    protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
+-                                              MinimumAgeRequirement requirement)
++                                              MinimumAgeAuthorizeAttribute requirement)
+    {
+        // Log as a warning so that it's very clear in sample output which authorization
+        // policies(and requirements/handlers) are in use
+        _logger.LogWarning("Evaluating authorization requirement for age >= {age}",
+                                                                    requirement.Age);
```

Comment:
Can we truncate this code block so that it only shows the method signature? The doc takes a long time to scroll through at the moment.